prometheus_client/exposition.py

```diff
@@ -118,11 +118,29 @@ def prometheus_app(environ, start_response):
         accept_header = environ.get('HTTP_ACCEPT')
         accept_encoding_header = environ.get('HTTP_ACCEPT_ENCODING')
         params = parse_qs(environ.get('QUERY_STRING', ''))
-        if environ['PATH_INFO'] == '/favicon.ico':
+
+        if environ['REQUEST_METHOD'] == 'OPTIONS':
+            if environ['PATH_INFO'] == '/*':
+                status = '200 OK'
+                headers = [('Allow', 'OPTIONS,GET')]
+                output = b''
+            else:
+                status = '404 Resource not found: '
+                headers = [('', '')]
+                output = '# HTTP {}: {}; use /*\n'.format(status, environ['PATH_INFO']).encode()
+        elif environ['REQUEST_METHOD'] != 'GET':
+            status = '405 Method Not Allowed'
+            headers = [('Allow', 'OPTIONS,GET')]
+            output = '# HTTP {}: {}; use OPTIONS or GET\n'.format(status, environ['REQUEST_METHOD']).encode()
+        elif environ['PATH_INFO'] == '/favicon.ico':
             # Serve empty response for browsers
             status = '200 OK'
             headers = [('', '')]
             output = b''
+        elif environ['PATH_INFO'].strip('/') not in ('', 'metrics'):
```

Comment:
This is the breaking change I am most worried about, some users might be using non-standard paths, or even just `/` to get their metrics as that all works right now and we don't require `/metrics`. What would you think of omitting this check?

Reply:
I agree with this. Even though `/metrics` is the convention, it's not enforced. e.g. for the client_golang, one can specify anything when they register a handler https://github.com/prometheus/client_golang/blob/e133e490296d2ff915bfc23cdec87c235ad36ef3/examples/simple/main.go#L43

Reply:
I can move the test fix into a aeparate PR.

Reply:
I was also unsure about the path checking and I can remove it.

Reply:
I have now updated the PR to remove the path check for GET. For consistency, I have also removed the path check for OPTIONS.

@kakkoyun @csmarchbanks Could you please review again, and merge if you agree?